src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -174,65 +175,78 @@ protected override void InitializeWorker(CompilationStartAnalysisContext context
+                        if (methodsWithSameNameAsTargetMethod.HasMoreThan(1))
+                        {
+                            var correctOverloads = methodsWithSameNameAsTargetMethod.GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(targetMethod, iformatProviderType).ToList();
+
+                            // If there are two matching overloads, one with CultureInfo as the first parameter and one with CultureInfo as the last parameter,
+                            // report the diagnostic on the overload with CultureInfo as the last parameter, to match the behavior of FxCop.
+                            var correctOverload = correctOverloads.FirstOrDefault(overload => overload.Parameters.Last().Type.Equals(iformatProviderType)) ?? correctOverloads.FirstOrDefault();
+
+                            // Sample message for IFormatProviderAlternateRule: Because the behavior of Convert.ToInt64(string) could vary based on the current user's locale settings,
+                            // replace this call in IFormatProviderStringTest.TestMethod() with a call to Convert.ToInt64(string, IFormatProvider).
+                            if (correctOverload != null)
+                            {
+                                oaContext.ReportDiagnostic(
+                                    invocationExpression.Syntax.CreateDiagnostic(
+                                        rule,
+                                        targetMethod.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
+                                        oaContext.ContainingSymbol.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
+                                        correctOverload.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat)));
+                            }
+                        }
                     }
                 }
                 #endregion
 
                 #region "UICultureStringRule & UICultureRule"
-                IEnumerable<int> IformatProviderParameterIndices = GetIndexesOfParameterType(targetMethod, iformatProviderType);
-                foreach (var index in IformatProviderParameterIndices)
                 {
-                    var argument = invocationExpression.Arguments[index];
+                    var rule = targetMethod.ReturnType.Equals(stringType) ?
+                        UICultureStringRule :
+                        UICultureRule;
 
-                    if (argument != null && currentUICultureProperty != null &&
-                        installedUICultureProperty != null && currentThreadCurrentUICultureProperty != null)
+                    if (!oaContext.Options.IsConfiguredToSkipAnalysis(rule, targetMethod, oaContext.ContainingSymbol, oaContext.Compilation))
```

Comment:
[cosmetic] Could you revert the condition to reduce nesting?

Reply:
Would you please elaborate on what you mean? It's not obvious to me. I understand what you want but not how to do it actually.

Reply:
@Evangelink This probably can't be done without `goto`? We don't bail out if it's configured to skip analysis since there are more rules to check.

However, I actually think that all `IsConfiguredToSkipAnalysis` calls will return the same result since all the 4 rules use the same ID?

In this case, I'll consider updating `IsConfiguredToSkipAnalysis` to take a rule id instead. But that probably should be in a different PR. Is there anything I'm missing?

Reply:
> However, I actually think that all `IsConfiguredToSkipAnalysis` calls will return the same result since all the 4 rules use the same ID?

This is something I'm interested in too.